python/ray/serve/tests/test_deploy.py

```diff
@@ -653,5 +653,79 @@ class DClass:
     assert D.version is None
 
 
+def test_deploy_multiple_apps_batched(serve_instance):
+    @serve.deployment
+    class A:
+        def __call__(self):
+            return "a"
+
+    @serve.deployment
+    class B:
+        def __call__(self):
+            return "b"
+
+    serve.run_many(
+        [
+            serve.RunTarget(A.bind(), name="a", route_prefix="/a"),
+            serve.RunTarget(B.bind(), name="b", route_prefix="/b"),
+        ]
+    )
+
+    assert serve.get_app_handle("a").remote().result() == "a"
+    assert serve.get_app_handle("b").remote().result() == "b"
+
+    assert requests.get("http://localhost:8000/a").text == "a"
+    assert requests.get("http://localhost:8000/b").text == "b"
+
+
+def test_redeploy_multiple_apps_batched(serve_instance):
+    @serve.deployment
+    class A:
+        def __call__(self):
+            return "a", os.getpid()
+
+    @serve.deployment
+    class V1:
+        def __call__(self):
+            return "version 1", os.getpid()
+
+    @serve.deployment
+    class V2:
+        def __call__(self):
+            return "version 2", os.getpid()
+
+    serve.run_many(
+        [
+            serve.RunTarget(A.bind(), name="a", route_prefix="/a"),
+            serve.RunTarget(V1.bind(), name="v", route_prefix="/v"),
+        ]
+    )
+
+    a1, pida1 = serve.get_app_handle("a").remote().result()
+
+    assert a1 == "a"
+
+    v1, pid1 = serve.get_app_handle("v").remote().result()
+
+    assert v1 == "version 1"
+
+    serve.run_many(
+        [
+            serve.RunTarget(V2.bind(), name="v", route_prefix="/v"),
+        ]
+    )
+
+    v2, pid2 = serve.get_app_handle("v").remote().result()
+
+    assert v2 == "version 2"
+    assert pid1 != pid2
+
+    # Redeploying "v" should not have affected "a"
+    a2, pida2 = serve.get_app_handle("a").remote().result()
+
+    assert a1 == a2
+    assert pida1 == pida2
```

Comment:
Could you also add a test to `test_local_testing_mode.py`?

Reply:
Can do - anything in particular that I should be checking there, or just an equivalent of these tests but with `_local_testing_mode=True` in the `run` calls?